### PR TITLE
tikv-ctl: remove pessimistic lock with invalid for_update_ts in recover-mvcc

### LINKED ISSUE #7981: Make `tikv-ctl recover-mvcc`be compatible with pessimistic transactions
## Development Task
Now `tikv-ctl recover-mvcc` can only work in pure optimistic transactions, it should work in mixed transaction mode.

### PR DESCRIPTION
**Pull request by sticnarf**
### What problem does this PR solve?

Issue Number: close #7981 

Problem Summary:

`tikv-ctl recover-mvcc` is not adapted for pessimistic transactions.

### What is changed and how it works?

What's Changed:

This PR refines the "recover-mvcc" tool. It can now remove pessimistic locks whose for_update_ts is less than the commit_ts of the latest write record.

### Related changes

- Need to cherry-pick to the release branch

### Check List 

Tests 

- Unit test

### Release note 

`tikv-ctl recover-mvcc` removes invalid pessimistic locks.

### CODE REVIEW

**Comment by sticnarf** [comment]
@youjiali1995 Actually I think current tikv-ctl can work under pessimistic transactions. The only thing pessimistic transactions add to TiKV is pessimistic locks. Remaining invalid pessimistic locks can be simply resolved by clients.

File: src/server/debug.rs [hunk 1]
```diff
@@ -970,13 +970,21 @@ impl MvccChecker {
             // If lock exists, check whether the records in DEFAULT and WRITE
             // match it.
             if let Some(ref l) = lock {
-                // All write records' ts should be less than lock's ts.
+                // All write records's ts should be less than the for_update_ts (if the
+                // lock is a pessimistic lock) or the ts of the lock.
```
Comment (MyonKeminta) [inline src/server/debug.rs:974]: I think when a pessimistic lock is replaced with a normal lock, its start_ts will be kept. So when pessimistic transaction is enabled, the non-pessimistic locks can also have less ts than the latest write does.

Comment (youjiali1995) [reply src/server/debug.rs:974]: Oh, yes. Should use `max(l.ts, l.for_update_ts)`.

**Comment by youjiali1995** [comment]
@MyonKeminta PTAL

**Pull request merged**